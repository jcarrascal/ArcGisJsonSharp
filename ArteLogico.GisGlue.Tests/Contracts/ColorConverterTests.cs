using ApprovalTests;
using ApprovalTests.Reporters;
using ArteLogico.GisGlue.Contracts;
using Newtonsoft.Json;
using NUnit.Framework;
using System.IO;

namespace ArteLogico.GisGlue.Tests.Contracts
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class ColorConverterTests
    {
        [Test]
        public void ReadJsonWriteJson_WhenSerializingAColor_ThenTheFormatMatchesTheExample()
        {
            string example = "[ 108, 151, 203, 0 ]";
            Color color;
            var serializer = JsonHelper.CreateSerializer();

            string result;
            using (var writer = new StringWriter())
            using (var jsonWriter = new JsonTextWriter(writer))
            using (var reader = new StringReader(example))
            using (var jsonReader = new JsonTextReader(reader))
            {
                jsonWriter.Formatting = Formatting.Indented;
                color = serializer.Deserialize<Color>(jsonReader);
                serializer.Serialize(jsonWriter, color);
                result = writer.ToString();
            }

            Approvals.Verify(color);
            Assert.That(result, Is.EqualTo(example));
        }
    }
}