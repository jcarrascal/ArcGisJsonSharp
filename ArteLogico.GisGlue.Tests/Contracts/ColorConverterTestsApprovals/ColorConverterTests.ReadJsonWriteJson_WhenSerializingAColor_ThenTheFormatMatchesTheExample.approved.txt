ArteLogico.GisGlue.Contracts.Color [108, 151, 203, 0]